### PR TITLE
DRAFT: Working on tab's testing. Test with cassandra, almost done.

### PR DESCRIPTION
**Pull request by tissarni**
Test Create/Update with condition with/without ID seems don't work. Can't test Get all tabs cause of the first problem.

Co-authored-by: Christophe Hamerling <christophe.hamerling@gmail.com>

### CODE REVIEW

File: twake/backend/node/src/services/channels/services/tab/service.ts [hunk 1]
```diff
@@ -45,6 +47,22 @@ export class Service implements TabService {
...
+
+    const userMember = await this.members.get({
+      channel_id: pk.channel_id,
+      company_id: pk.company_id,
+      workspace_id: pk.workspace_id,
+      user_id: context.user.id,
+    });
+
+    if (!userMember) {
+      throw CrudExeption.badRequest("User is not channel member");
```
Comment (chamerling) [inline twake/backend/node/src/services/channels/services/tab/service.ts:63]: Replace with "User is now allowed to create tab"

File: twake/backend/node/src/services/channels/services/tab/service.ts [hunk 2]
```diff
@@ -121,6 +151,17 @@ export class Service implements TabService {
   onDeleted(channel: Channel, tab: ChannelTab): void {
     console.log("Tab deleted", tab);
   }
+
+  async userMember(pk: ChannelTabPrimaryKey, context: ChannelExecutionContext): Promise<ChannelMember>{
```
Comment (chamerling) [inline twake/backend/node/src/services/channels/services/tab/service.ts:155]: If you want to have a function like this, you can, but you have to name it correctly: `userIsChannelMember`, and you can return the result directly like:

```
async userIsChannelMember(user: string, channel: Channel): Promise<boolean>{
  return !!(await this.members.get({ CHANNEL }));
}
```

File: twake/backend/node/test/http/tabs.http [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+### Delete a tab
+@deleteId = {{createTab.response.body.resource.id}}
+
+DELETE {{tabsURL}}/{{channelId}}/tabs/{{deleteId}}
+Authorization: Bearer {{authToken}}
+
+
+
+
+
```
Comment (chamerling) [inline twake/backend/node/test/http/tabs.http:90]: TO be removed

**Comment by chamerling** [comment]
How did you tested it? 
1. Does it work on mongo?
2. Does it work on scylladb?

**Comment by tissarni** [comment]
> How did you tested it?
> 
> 1. Does it work on mongo?
> 2. Does it work on scylladb?

I tested it with Mongo and Scylladb both started manually.

**Comment by RomaricMourgues** [comment]
Replaced by #789

**Pull request closed without merging**